NSrtm.Core/Pgm/GridGraph/PgmGridGraphInMemory.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using JetBrains.Annotations;
+
+namespace NSrtm.Core.Pgm.GridGraph
+{
+    public sealed class PgmGridGraphInMemory : IPgmGridGraph
+    {
+        private readonly UInt16[] _scaledUndulation;
+        private readonly PgmDataDescription _pgmParameters;
+
+        [CLSCompliantAttribute(false)]
```

Comment:
I can change the name to GeoidUndulationGrid . NGA uses the term to describe their files. IMO this is descriptive enough . I am waiting for your opinion . 

At this point we don't have a provider that is why i do not want to speak about it without its full vision.